### ISSUE TITLE
socialMinerBaseUrl is set to localhost 

### ISSUE THREAD

**Issue by sylvesterlenihan** [CiscoDevNet/socialminer-sample-code#10]
In line 36 of chat.html, socialMinerBaseUrl is set to localhost, meaning the sample code will only work if ran on socialminer server.

`socialMinerChat.init(window.location.protocol + "//" + window.location.host, "........`

**Comment by umnagendra** [comment]
Hi @sylvesterlenihan,

The `socialMinerBaseUrl` argument is indeed set to localhost (rather, it is set to the server where the node app is running and serving the `chat.html`).

However, when you launch this sample, you do provide the IP/hostname of your SocialMiner server

`node app.js --socialminer <SocialMiner_IP_or_HOSTNAME>`

This is used inside the node app (`app.js`) which is acting as a proxy for SocialMiner. In this node app, the request from the web page is handled and forwarded to SocialMiner server.

See [line 48](https://github.com/CiscoDevNet/socialminer-sample-code/blob/master/customer-chat/app.js#L48) in `app.js` and you'll get it.

Rest assured, the sample works fine no matter where you deploy it - in fact, it should not be deployed inside SocialMiner server.

Let me know if you have any further questions.
I will close this issue once you are clear.

**Comment by sylvesterlenihan** [comment]
Hi,
Thanks for reply.
I am launching he app with the command 

`node app.js --socialminer <SocialMiner_IP_or_HOSTNAME>`

but I am getting a 404 on the post on the post to /ccp/chat/ as it was going to 

`http://localhost/ccp/chat/`

the following workaround in (customer_chat.js) fixed the issue 
      
init: function (socialMinerBaseUrl, chatFeedRefUrl)
        {
            console.log('social miner is ' +socialMinerBaseUrl);
            chatUrl =  "http://_SocialMiner_IP_/ccp/chat/";
            //chatUrl = socialMinerBaseUrl + "/ccp/chat/";
            feedRefUrl = chatFeedRefUrl;
        },

**Comment by umnagendra** [comment]
@sylvesterlenihan That is indeed interesting. I ran the sample at my end **exactly** as recommended in the README, and I was able to do a successful end-2-end web chat with SocialMiner and CCX.

The node app is just a proxy - the web page always communicates only with the node app.
Then, the node app will in-turn communicate with SocialMiner. (That is why the SocialMiner IP is provided to the node app)

If you are getting a 404 NOT FOUND from your web page to the node app (localhost), that possibly means there is some issue reaching your node app on the port (it might be running on another port?)

Let me know if you are still having an issue - we can quickly sync-up over webex.
